example/example/tests/test_views.py

```diff
@@ -39,3 +41,63 @@ def test_proxy_s3_view(api_client, mock_data_client):
     url = f"{reverse(view_name)}?{invalid_url_params}"
     response = api_client.get(url)
     assert response.status_code == status.HTTP_500_INTERNAL_SERVER_ERROR
+
+
+@pytest.mark.django_db
+class TestUserTracking:
+    def test_submitting_non_list(self, api_client):
+        """
+        Ensure data posted to view is an array/list.
+        """
+        log_data = {"content": {"key": "Value 1", }, "tags": ["dataset"]}
+
+        url = reverse("log-tracking")
+
+        response = api_client.post(url, log_data, format="json")
+
+        assert not status.is_success(response.status_code)
+        assert response.data == "Must supply an array of JSON objects in request"
+
+    def test_tracking_missing_content(self, api_client):
+        """
+        Ensure data posted to view has object called 'content'.
+        """
+        log_data = [{"key": "Value 1"}]
+
+        url = reverse("log-tracking")
+
+        response = api_client.post(url, log_data, format="json")
+
+        assert not status.is_success(response.status_code)
+        assert response.data == "Log Record must contain key 'content' of JSON to be logged"
+
+    def test_tracking_features(self, api_client, astrosat_settings):
+        """
+        Ensure correct data posted is handled.
+        """
+        # make sure logging is enabled...
+        astrosat_settings.enable_db_logging = True
+        astrosat_settings.save()
+
+        log_data = [
+            {
+                "content": {
+                    "key": "Value 1",
+                },
+                "tags": ["dataset"]
+            }, {
+                "content": {
+                    "key": "Value 2",
+                },
+                "tags": ["dataset"]
+            }
+        ]
+
+        url = reverse("log-tracking")
+
+        assert DatabaseLogRecord.objects.count() == 0
+
+        response = api_client.post(url, log_data, format="json")
+
+        assert status.is_success(response.status_code)
+        assert DatabaseLogRecord.objects.count() == 2
```

Comment:
Might also want to check:

```
assert DatabaseLogTag.objects.count() == 1
```

To make sure that one and only one tag was created.

Reply:
So another test, but with just one log you mean?

Reply:
No, the same test.  But you want to check that for the first element of data the code created a new `DatabaseLogTag` called "dataset" and for the second element of data the code re-used that same `DatabaseLogTag` instance.

Reply:
Ah, right, sorry, didn't pick up on you meaning the `tag`, yes that does make perfect sense.